landlab/components/dimensionless_discharge/dimensionless_discharge.py

```diff
@@ -0,0 +1,196 @@
+     >>> dd = DimensionlessDischarge(watershed_grid)
+     >>> dd.run_one_step()
+     >>> print(watershed_grid.at_node['dimensionless_discharge'])
+     [ 0.55372743  0.55372743  0.55372743  0.55372743  0.55372743
+             0.55372743  0.55372743  0.55372743  0.55372743]
+
+
+
+     References
+     ----------
+     Tang, H., McGuire, L. A., Rengers, F. K., Kean, J. W., Staley,
+     D. M., & Smith, J. B. (2019). Developing and Testing Physically
+     Based Triggering Thresholds for Runoff-Generated Debris Flows.
+     Geophysical Research Letters, 46(15), 8830–8839.
+     https://doi.org/10.1029/2019GL083623
+
+    """
+
+    _name = "DimensionlessDischargeModel"
+
+    _unit_agnostic = False
+
+    _info = {
+        "dimensionless_discharge": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "dimensionless_discharge_above_threshold": {
+            "dtype": bool,
+            "intent": "out",
+            "optional": False,
+            "units": "none",
+            "mapping": "node",
+            "doc": "Dimensionless discharge value for a stream segment.",
+        },
+        "flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "flux in each stream segment",
+        },
+        "d50": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "soil grain size average in stream segment",
+        },
+        "dem_values": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "elevation stream segment",
+        },
+    }
+
+    def __init__(self, grid, soil_density=1330, water_density=997.9, C=12.0, N=0.85):
+        """Initialize the DimensionlessDischarge.
+
+        Parameters
+        ----------
+        soil_density : float, optional (defaults to 1330)
+            density of soil (kg/m^3)
+        water_density : float, optional (defaults to 997.9)
+            density of water (kg/m^3)
+        C : float, optional (defaults to 12.0)
+            Numerator of the debris flow threshold equation; Empirically
+            derived constant (See Tang et al. 2019)
+        N : float, optional (defaults to 0.85)
+            Exponent for slope in the denominator of the debris flow
+            threshold equation; Empirically derived constant (See Tang
+            et al. 2019)
+        """
+
+        super().__init__(grid)
+
+        # Store parameters
+        self._soil_density = soil_density
+        self._C = [C] * grid.number_of_nodes
+        self._N = [N] * grid.number_of_nodes
```

Comment:
If slopes will be calculated within *run_one_step*, then both *C* and *N* will need to be saved. Since *numpy* will broadcast both to the correct shape in the *dimensionless_discharge_threshold_value calculation*, you can just save the passed values.
```python
self._c = C
self._n = N
```

Reply:
Updated.